controllers/provider-aws/pkg/webhook/controlplane/ensurer.go

```diff
@@ -205,3 +204,42 @@ func (e *ensurer) EnsureKubernetesGeneralConfiguration(ctx context.Context, data
 	*data = buf.String()
 	return nil
 }
+
+func appendUniqueUnit(units *[]extensionsv1alpha1.Unit, unit extensionsv1alpha1.Unit) {
+	for _, un := range *units {
+		if un.Name == unit.Name {
+			return
+		}
+	}
+	*units = append(*units, unit)
+}
+
+// EnsureAdditionalUnits ensures that additional required system units are added.
+func (e *ensurer) EnsureAdditionalUnits(ctx context.Context, units *[]extensionsv1alpha1.Unit) error {
```

Comment:
Can you add a test for this in https://github.com/gardener/gardener-extensions/blob/master/controllers/provider-aws/pkg/webhook/controlplane/ensurer_test.go?